internal/resources/gitrepository/resource_git_repository.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package gitrepository
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common"
+	commonscope "github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common/scope"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/gitrepository/scope"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/gitrepository/spec"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/gitrepository/status"
+)
+
+func ResourceGitRepository() *schema.Resource {
```

Comment:
There same question with gitrepository and kustomization resources , should we really define them as separate resources ?

Reply:
I've rebased it, so git repository resource is not there in this MR